jobs/kpi-forecasting/kpi_forecasting/tests/test_utils.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+
+from kpi_forecasting.utils import parse_end_date, previous_period_last_date
+
+
+def test_parse_end_date_with_date():
+
+    dt = "2024-01-01"
+    parsed_date = parse_end_date(dt)
+
+    assert dt == parsed_date
+
+
+def test_parse_end_date_with_none():
+
+    dt = None
+    parsed_date = parse_end_date(dt)
+
+    assert parsed_date.date() == datetime.datetime.utcnow().date()
+
+
+def test_parse_end_date_prev_year():
+
+    dt = "last complete year"
+    parsed_date = parse_end_date(dt)
+
+    yr = datetime.datetime.utcnow().date()
+
+    assert parsed_date == datetime.date(yr, 12, 1)
```

Comment:
For my own knowledge, when do the tests run? I assumed it was upon commit as part of the 17 checks run in github...but apparently not since this code shouldn't run at all: `TypeError: 'datetime.date' object cannot be interpreted as an integer`

Reply:
They're currently not set up to run. It'll be included in the `ci_job.yaml` steps eventually, when they are going to run, as part of the Docker run command. I'll request help from DE when setting those up, though, to make sure it's done correctly.